lib/snowflake/index.js

```diff
@@ -3,6 +3,34 @@ const snowflake = require('snowflake-sdk')
 const Log = require('@kelchy/log')
 
 class Snowflake extends EventEmitter {
+  /**
+   *
+   * @param {string} connStr - format: "snowflake://<username>:<password>@<account>.snowflakecomputing.com/?role=<role>&warehouse=<warehouse>&database=<database>&schema=<schema>"
+   * @returns snowflake credentials obj
+   */
+  static parseConnStr (connStr) {
+    const url = new URL(connStr)
+    const query = new URLSearchParams(url.search)
+
+    // extract account from host
+    let account = null
+    const hostMatch = url.host.match(/^(.*?).snowflakecomputing.com/)
+    if (hostMatch && hostMatch.length > 0) {
+      account = hostMatch[1]
+    }
+
+    // URL Encoding (percent encoding) was done automatically, need to decode to get the real characters
+    const password = decodeURIComponent(url.password)
+
+    const username = url.username
+    const role = query.get('role')
+    const warehouse = query.get('warehouse')
+    const database = query.get('database')
+    const schema = query.get('schema')
+
+    return { account, username, password, role, warehouse, database, schema }
+  }
+
   constructor (snowflakeOpts = {}, options = {}) {
```

Comment:
i suggest changing this to uri as the first argument and options as the second
don't worry about backwards compatibility now as we know which projects are using this currently
handle the uri the moment it goes into the constructor just as how we handle this in redis, mongo, mysql, rmq, etc

Reply:
Agreed. We should use the simplest way (i.e providing just a URI for connection). Will make the changes